shopping/src/main/java/ch/ddd/foodwatch/shopping/infrastructure/persistence/ShoppingRepository.java

```diff
@@ -1,15 +1,43 @@
 package ch.ddd.foodwatch.shopping.infrastructure.persistence;
 
-import org.springframework.stereotype.Component;
-
 import ch.ddd.foodwatch.shopping.domain.ShoppingList;
 import ch.ddd.foodwatch.shopping.domain.ShoppingListId;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.EntityManager;
+import java.util.List;
+
+import static java.util.stream.Collectors.toList;
 
-@Component
+@Repository
 public class ShoppingRepository {
```

Comment:
Hi @kullmanp,
is there a reason why you don't use Spring Data JPA Repository? Personally I don't like the jpa, i prefer the spring data jdbc approach. What is your opinion?

Reply:
I don't use the spring magic repositories because they have the entity classes in the interface. I want to keep classes with "@Entity" hidden in the repo implementation and want to expose only domain classes.

spring data jdbc vs spring data jpa -- I don't really care. Jpa is simpler I think for this one: I get schema definition for free and I don't have to write sql.

Reply:
@kullmanp you can inject the spring magic class via delegation to our repository ;-). With spring data jdbc you don´t need to write sql ;-). But I like the approach because it is based on aggregates https://spring.io/projects/spring-data-jdbc